assets/wizards/popups/views/campaigns/index.js

```diff
@@ -0,0 +1,301 @@
+};
+
+const dataForCampaignId = ( id, campaigns ) =>
+	campaigns.reduce( ( acc, group ) => ( +id > 0 && +id === +group.term_id ? group : acc ), null );
+
+/**
+ * Campaign management screen.
+ */
+const Campaigns = props => {
+	const {
+		campaignId,
+		prompts = [],
+		campaigns = [],
+		manageCampaignGroup,
+		segments,
+		wizardApiFetch,
+		refetch,
+		duplicateCampaignGroup,
+		deleteCampaignGroup,
+		archiveCampaignGroup,
+		renameCampaignGroup,
+	} = props;
+
+	const [ popoverVisible, setPopoverVisible ] = useState();
+	const [ modalVisible, setModalVisible ] = useState();
+	const [ modalType, setModalType ] = useState();
+	const [ campaignName, setCampaignName ] = useState();
+	const [ inFlight, setInFlight ] = useState( false );
+
+	const history = useHistory();
+
+	const submitModal = modalText => {
+		if ( MODAL_TYPE_NEW === modalType ) {
+			createTerm( modalText );
+		} else if ( MODAL_TYPE_RENAME === modalType ) {
+			renameCampaignGroup( campaignId, modalText );
+		} else if ( MODAL_TYPE_DUPLICATE === modalType ) {
+			duplicateCampaignGroup( campaignId, modalText );
+		}
+		setModalVisible( false );
+	};
+
+	const createTerm = term => {
+		setPopoverVisible( false );
+		setInFlight( true );
+		wizardApiFetch( {
+			path: '/wp/v2/newspack_popups_taxonomy',
+			method: 'POST',
+			quiet: true,
+			data: {
+				name: term,
+				slug: term,
+			},
+		} )
+			.then( () => {
+				setInFlight( false );
+				setCampaignName( '' );
+				refetch();
+			} )
+			.catch( () => {
+				setInFlight( false );
+			} );
+	};
+
+	const activeCampaigns = campaigns.filter( ( { status } ) => 'archive' !== status );
+	const archivedCampaigns = campaigns.filter( ( { status } ) => 'archive' === status );
+	const campaignData = dataForCampaignId( campaignId, campaigns );
+
+	const valueForCampaignId = id => {
+		if ( 'unassigned' === id ) {
+			return {
+				key: 'unassigned',
+				name: __( 'Unassigned Prompts', 'newspack' ),
+			};
+		}
+
+		const data = dataForCampaignId( id, campaigns );
+
+		if ( data ) {
+			return {
+				key: data.term_id,
+				name: data.name,
+			};
+		}
+		return {
+			key: 'active',
+			name: __( 'Active Prompts', 'newspack' ),
+		};
+	};
+	return (
+		<Fragment>
+			<div className="newspack-campaigns__campaign-group__filter-group-wrapper">
+				<div className="newspack-campaigns__campaign-group__filter-group-actions">
+					<CustomSelectControl
+						label={ __( 'Campaigns', 'newspack' ) }
+						options={ [
+							{ key: 'active', name: __( 'Active Prompts', 'newspack' ) },
+							{ key: 'unassigned', name: __( 'Unassigned Prompts', 'newspack' ) },
+							{
+								key: 'header-campaigns',
+								name: __( 'Campaigns', 'newspack' ),
+								className: 'is-header',
+							},
+							...activeCampaigns.map( ( { term_id: id, name } ) => ( {
+								key: id,
+								name,
+								className: 'newspack-campaigns__campaign-group__select-control-group-item',
+							} ) ),
+							archivedCampaigns.length && {
```

Comment:
If there are no archived campaigns, `0` is passed among options to the `CustomSelectControl`, which is not a valid option. Here's a trick for inline conditional array elements:

```js
...( archivedCampaigns.length
	? [
			{
				key: 'header-archived-campaigns',
				name: __( 'Archived Campaigns', 'newspack' ),
				className: 'is-header',
			},
	  ]
	: [] ),
```

Reply:
Neat! https://github.com/Automattic/newspack-plugin/pull/833/commits/3dbf75e6ee52dc41c5e3a5d1dd78ee894710fb0d